src/resolve/resolvconf-compat.c

```diff
@@ -53,6 +53,8 @@ static int parse_nameserver(const char *string) {
 
                 if (strv_push(&arg_set_dns, word) < 0)
                         return log_oom();
+
+                TAKE_PTR(word);
```

Comment:
You have a point about `TAKE_PTR()` being unsafe when used as an argument to a function that can fail, but here it's not really necessary, `word = NULL;` is more straightforward, so I'd suggest sticking to that (just fixing the fact that this one didn't set it here.)

Reply:
Yeah, `word = NULL;` is sufficient. But using `TAKE_PTR()` is not bad. We have already used `TAKE_PTR()` in such cases. C.f. https://github.com/systemd/systemd/blob/ad4bc3352285f467f4ffa03c3171b19fa0a8758d/src/libsystemd/sd-resolve/sd-resolve.c#L1054

Reply:
I've changed to `word = NULL;`, as the author of the first commit is @filbranden.

Reply:
Thanks!

Yeah, maybe it's OK (the compiler will produce the same) but it's a bit weird seing it used when the value gets discarded, given the whole point of it is preserving the value...